### ISSUE TITLE
Bug inside method OneLogin_Saml2_Utils::getSelfRoutedURLNoQuery

### ISSUE THREAD

**Issue by michalwolek** [onelogin/php-saml#384]
Hello.

I found bug inside method [OneLogin_Saml2_Utils::getSelfRoutedURLNoQuery](https://github.com/onelogin/php-saml/blob/master/lib/Saml2/Utils.php#L621-L643). When i run demo on url: `http://example.com/sso/demo1/?sso`, this method change `/sso/demo1/?sso` to `//demo1/` instead of `/sso/demo1/`.
I suggest to use preg_replace
```php
$route = preg_replace(
    '/^(.*)\?' . $_SERVER['QUERY_STRING'] . '.*$/',
    '$1',
    $_SERVER['REQUEST_URI']
);
```
instead of str_replace
https://github.com/onelogin/php-saml/blob/bedde3ee3965254b0c1faf3cf18c1fb564314f0d/lib/Saml2/Utils.php#L629-L632

Best regards,
Michal.

**Comment by pitbulk** [comment]
I will review and let you know

**Comment by pitbulk** [comment]
Fixed 36aa9554335b2eb07f5add4929a5d4c35499bc19
